py/map.c

```diff
@@ -126,6 +126,15 @@ STATIC void mp_map_rehash(mp_map_t *map) {
     m_del(mp_map_elem_t, old_table, old_alloc);
 }
 
+#if MICROPY_OPT_MAP_CACHING
+uint8_t map_lookup_cache[128];
+#define MAP_CACHE_OFFSET ((((uintptr_t) + (uintptr_t)index) >> 2) % sizeof(map_lookup_cache))
```

Comment:
I think this first `uintptr_t` is missing something that it casts, maybe `map` or `map->table`?

Reply:
Originally I had this using `(map+index)>>2`, but changed it to just use `index>>2` because it's overall faster. Accidentally left behind the no-op cast. I've updated the branch, and addressed the other comment about the macro args.

Here's the diff FWIW of adding back `map` to the expression. Makes some things marginally better, but big hit on pystone and raytrace.

```
$ ./run-perfbench.py -s ~/2.cache-index-pybv11 ~/2.cache-mapindex-pybv11
diff of scores (higher is better)
N=100 M=100              /home/jimmo/2.cache-index-pybv11 -> /home/jimmo/2.cache-mapindex-pybv11         diff      diff% (error%)
bm_asyncio.py                113.73 ->     113.43 :      -0.30 =  -0.264% (+/-0.01%)
bm_chaos.py                  348.59 ->     340.54 :      -8.05 =  -2.309% (+/-0.01%)
bm_fannkuch.py                78.53 ->      78.20 :      -0.33 =  -0.420% (+/-0.00%)
bm_fft.py                   2563.02 ->    2597.15 :     +34.13 =  +1.332% (+/-0.00%)
bm_float.py                 5544.62 ->    5440.36 :    -104.26 =  -1.880% (+/-0.02%)
bm_hexiom.py                  42.73 ->      42.74 :      +0.01 =  +0.023% (+/-0.00%)
bm_nqueens.py               4429.89 ->    4454.66 :     +24.77 =  +0.559% (+/-0.00%)
bm_pidigits.py               651.39 ->     649.83 :      -1.56 =  -0.239% (+/-0.34%)
misc_aes.py                  423.37 ->     428.64 :      +5.27 =  +1.245% (+/-0.01%)
misc_mandel.py              3610.82 ->    3624.53 :     +13.71 =  +0.380% (+/-0.01%)
misc_pystone.py             2287.95 ->    2185.46 :    -102.49 =  -4.480% (+/-0.01%)
misc_raytrace.py             362.17 ->     341.62 :     -20.55 =  -5.674% (+/-0.01%)
```

And while we're at it, here's the latest baseline->PR after rebasing (similar to the first table in https://github.com/micropython/micropython/pull/7680#issuecomment-901681780)

```
$ ./run-perfbench.py -s ~/2.cache-baseline-pybv11 ~/2.cache-index-pybv11
diff of scores (higher is better)
N=100 M=100              /home/jimmo/2.cache-baseline-pybv11 -> /home/jimmo/2.cache-index-pybv11         diff      diff% (error%)
bm_asyncio.py                111.34 ->     113.73 :      +2.39 =  +2.147% (+/-0.01%)
bm_chaos.py                  311.32 ->     348.59 :     +37.27 = +11.972% (+/-0.02%)
bm_fannkuch.py                77.82 ->      78.53 :      +0.71 =  +0.912% (+/-0.00%)
bm_fft.py                   2539.05 ->    2563.02 :     +23.97 =  +0.944% (+/-0.00%)
bm_float.py                 4955.56 ->    5544.62 :    +589.06 = +11.887% (+/-0.00%)
bm_hexiom.py                  35.52 ->      42.73 :      +7.21 = +20.298% (+/-0.00%)
bm_nqueens.py               4201.40 ->    4429.89 :    +228.49 =  +5.438% (+/-0.00%)
bm_pidigits.py               646.91 ->     651.39 :      +4.48 =  +0.693% (+/-0.17%)
misc_aes.py                  382.97 ->     423.37 :     +40.40 = +10.549% (+/-0.01%)
misc_mandel.py              3091.74 ->    3610.82 :    +519.08 = +16.789% (+/-0.01%)
misc_pystone.py             1948.00 ->    2287.95 :    +339.95 = +17.451% (+/-0.01%)
misc_raytrace.py             312.45 ->     362.17 :     +49.72 = +15.913% (+/-0.01%)
```